trussworks/react-uswds
# feat: File Input component

# Summary

Implements the [USWDS File Input component](https://designsystem.digital.gov/components/form-controls/#file-input).

There are some tests that aren't 100% accurate, because simulating a drop event doesn't seem to accurately trigger the onChange handler for a file input. I left outstanding comments in those cases and maybe they can be improved upon in the future.

## Related Issues or PRs

Closes #341 

## How To Test

**Note:** The FileInput requires updating to USWDS 2.8.1, which is included in this PR. Please make sure you run `yarn install` before testing after you check out this branch.

Test that you can use the input in a multipart form to upload file(s), and the file type validation works as expected.

If you want to handle uploading async on the change event, there is an `onChange` prop exposed you can use to pass in a custom handler. Note that the FileInput component will still display dropped files, and you may want to force a re-render to clear the value after uploading.

### Screenshots (optional)

![image](https://user-images.githubusercontent.com/2723066/108895482-3dcb0380-75d9-11eb-8de8-6287fa57fbb8.png)

![image](https://user-images.githubusercontent.com/2723066/108895494-43c0e480-75d9-11eb-90d1-5b7211972088.png)

![image](https://user-images.githubusercontent.com/2723066/108895526-50453d00-75d9-11eb-98d3-d98b4a9915b0.png)

![image](https://user-images.githubusercontent.com/2723066/108895547-563b1e00-75d9-11eb-9aa2-382769f8ee0b.png)
